### PR TITLE
Cleanup README; format samples

### PR DESCRIPTION
**Pull request by bartelink**
Thanks once again for bring this to the world; I've been using the skin [with some success](https://github.com/jet/equinox/pull/213).

That makes me an expert so I want to change all the things ;) For instance, I think being able to assign friendly name aliases to elements should be part of a standard sampleset, but I'm ok with you overruling and asking for aliases to be removed etc.

Please feel free to push back on some of the changes as I may well be missing the point etc

I've yet to run the build:
- it seems to need a .gitignore and/or otherwise deal with `.DS_Store` files (build chokes on it)
- when I did run it, the conflict with my removal of the `/` in the `/src/c4.puml` results in avoidable diffs in the output

### CODE REVIEW

File: src/c4.puml [hunk 1]
```diff
@@ -1,13 +1,13 @@
-!include /general/general.puml
-!include /general/legend.puml
-!include /general/notes.puml
+!include general/general.puml
```
Comment (bartelink) [inline src/c4.puml:1]: this was preventing me from editing with vscode - as mentioned in the overview, the build script will need to be adjusted to fit

File: README.md [hunk 2]
```diff
@@ -30,12 +31,12 @@ Just below `@startuml`, and then use the following components with the specified
 | C4 artifact | PlantUml | Stereotype for internal | Stereotype for external |
 |---|---|---|---|
 | Person | `actor` | `<<Internal>>` | `<<External>>` |
```
Comment (bartelink) [inline README.md:33]: is it feasible/ does it make sense to have `<<Person>>` and `<<External Person>>` aliases? I was using those, following the general namign scheme, and ending up with no skinning (I didnt think too deeply and never went back to RTFM)

File: README.md [hunk 3]
```diff
@@ -30,12 +31,12 @@ Just below `@startuml`, and then use the following components with the specified
...
-| System | `frame` | `<<Internal System>>`, <br> `<<Internal>>`, <br> or `<<System>>` | `<<External System>>` <br> or `<<External>>` |
-| System | `rectangle` | `<<Internal System>>` <br> or `<<System>>` | `<<External System>>` |
-| Containers | `component` | `<<Internal Container>>`, <br> `<<Internal>>`, <br> or `<<Container>>` | `<<External Container>>` <br> or `<<External>>` |
-| Containers | `rectangle` | `<<Internal Container>>` <br>or `<<Container>>` | `<<External Container>>` |
-| Component | `rectangle` | `<<Internal Component>>` <br> or `<<Component>>` | `<<External Component>>` |
-| Database | `database` | `<<Internal Database>>` <br> or `<<Internal>>` |  `<<External Database>>` <br> or `<<External>>` |
+| System | `frame` | `<<System>>`, <br> `<<Internal System>>`, <br> or `<<Internal>>` | `<<External System>>` <br> or `<<External>>` |
```
Comment (bartelink) [inline README.md:34]: I'm not sure if this reorg truly helps, but I was confused by this ordering - my heuristic was:

> you can say Internal, but its superfluous; in general, <<X>> will have a matching <<X External>>

Here I'm trying to put the shortest, least confusing alias first - not the one with obvious redundancy I need to mentally exclude

File: README.md [hunk 4]
```diff
@@ -30,12 +31,12 @@ Just below `@startuml`, and then use the following components with the specified
...
+| System | `rectangle` | `<<System>>` <br> or `<<Internal System>>` | `<<External System>>` |
+| Containers | `component` | `<<Container>>`, <br> `<<Internal Container>>`, <br> or `<<Internal>>` | `<<External Container>>` <br> or `<<External>>` |
+| Containers | `rectangle` | `<<Container>>` <br>or `<<Internal Container>>` | `<<External Container>>` |
+| Component | `rectangle` | `<<Component>>` <br> or `<<Internal Component>>` | `<<External Component>>` |
+| Database | `database` | `<<Internal Database>>` <br> or `<<Internal>>` | `<<External Database>>` <br> or `<<External>>` |
```
Comment (bartelink) [inline README.md:39]: is `<<Database>>` not doable? it would be more consistent

File: README.md [hunk 5]
```diff
@@ -64,32 +65,24 @@ Reproduced (as closely as possible from diagrams from [the official C4 model web
...
 
 1. It is a *pleasure* to edit
 1. You don't need special tooling
 1. You can keep your domain models side-by-side with your code
 1. You can embed models in your markdown documents in almost all major editors
-1. Did I mention that it is an absolute pleasure to work with plantuml?
+1. Did I mention that it is an absolute pleasure to work with Plantuml?
 
-The only real disadvantage to plantuml is that positioning can sometimes be a bit hard.
+The only real disadvantage is that positioning can sometimes be [a bit tricky](https://stackoverflow.com/a/48735216).
```
Comment (bartelink) [inline README.md:83]: wanted to say black magic, but mentioned further down ;)

File: README.md [hunk 6]
```diff
@@ -125,28 +118,27 @@ All that is needed to start is to add the following line:
...
 
-    External -down-> Internal
-    Internal -right-> InternalPerson
-    InternalPerson -up-> ExternalPerson
+    External       -down->  Internal
+    Internal       -right-> InternalPerson
+    InternalPerson -up->    ExternalPerson
 
-    @end
+    @enduml
```
Comment (bartelink) [inline README.md:132]: assuming this had no specific purpose?

File: README.md [hunk 7]
```diff
@@ -210,11 +203,11 @@ All that is needed to start is to add the following line:
     interface InternalInterface <<Internal Container>>
     interface ExternalInterface <<External Container>>
 
-    InternalComponent -right-> ExternalComponent
-    InternalRectangle -right-> ExternalRectangle
+    InternalComponent  -right-> ExternalComponent
```
Comment (bartelink) [inline README.md:206]: I use camelCase names - that would make it easier to distinguish stereotype names from names when scanning - happy to apply if you think it makes sense ...

File: README.md [hunk 8]
```diff
@@ -267,17 +260,26 @@ All that is needed to start is to add the following line:
...
+    - [Canonical SO answer re layout black magic](https://stackoverflow.com/a/48735216)
 
 ## work outstanding
 
 1. Complete examples
 1. Add skinning for C4 Code diagrams 
 
+## Updates
+
+- [2019-04-12] : Added skinning for sequence diagrams. Add `!define SKIN_SEQUENCE_DIAGRAM` before the include statement. See [a short sample](./sample/sequence.puml)
```
Comment (bartelink) [inline README.md:272]: think these are best here - it feels like an ending marker when encountered up the page

File: README.md (hunk 2 above)
Comment (skleanthous) [reply README.md:33]: Hey @bartelink, what are you asking exactly? Do you mean have aliases with just a a single angle-brackets: `<`, and aliases with double ones `<<`?

File: README.md (hunk 3 above)
Comment (skleanthous) [reply README.md:34]: I was going with more specific to less specific for clarity. I'd prefer it the way it was to be quite honest, but I'm not sure about the original being really more clear to everyone, and given the rest of the work done in this PR, I'm happy to merge this too.

File: README.md (hunk 5 above)
Comment (skleanthous) [reply README.md:83]: Calling it tricky is an understatement imho. It is downright impossible in some cases. And there are bugs in PlantUML that make it impossible to enforce the layout you want in some more "exotic" cases.

Comment (skleanthous) [reply README.md:83]: I am happy with this change though :)

File: README.md (hunk 6 above)
Comment (skleanthous) [reply README.md:132]: This was just an error on my part 😬

File: README.md (hunk 7 above)
Comment (skleanthous) [reply README.md:206]: That would be nice yes. I just used PascalCase due to habit than any other reason.

File: README.md (hunk 8 above)
Comment (skleanthous) [reply README.md:272]: 👍

File: README.md (hunk 4 above)
Comment (skleanthous) [reply README.md:39]: I commented above about this.

**Review by skleanthous — APPROVED** [review]

**Comment by skleanthous** [comment]
Awesome contribution, thank you @bartelink! I'll merge it. Please provide some clarification re https://github.com/skleanthous/C4-PlantumlSkin/pull/14#discussion_r402619105 and we can discuss this specifically.

Also, sorry for not responding sooner; I've seen this and wanted tor review it sooner, but it was not possible. Again thank you for this contribution. I'll see what I can do about the build script.

File: README.md (hunk 2 above)
Comment (bartelink) [reply README.md:33]: sorry, didnt quote the angle brackets :(

File: README.md (hunk 4 above)
Comment (bartelink) [reply README.md:39]: added quoting to clarify comment...

File: README.md (hunk 3 above)
Comment (bartelink) [reply README.md:34]: I guess I'll have to have to use it more to reach a final conclusion; would have been fine with this aspect being rolled back either - resolving...

**Pull request merged**